Source/Core/UICommon/GameFile.cpp

```diff
@@ -720,31 +720,51 @@ GameFile::CompareSyncIdentifier(const NetPlay::SyncIdentifier& sync_identifier)
   if ((is_elf_or_dol ? m_file_size : 0) != sync_identifier.dol_elf_size)
     return NetPlay::SyncIdentifierComparison::DifferentGame;
 
-  const auto trim = [](const std::string& str, size_t n) {
-    return std::string_view(str.data(), std::min(n, str.size()));
-  };
-
-  if (trim(m_game_id, 3) != trim(sync_identifier.game_id, 3))
+  if (m_is_datel_disc != sync_identifier.is_datel)
     return NetPlay::SyncIdentifierComparison::DifferentGame;
 
-  if (m_disc_number != sync_identifier.disc_number || m_is_datel_disc != sync_identifier.is_datel)
+  if (m_game_id.size() != sync_identifier.game_id.size())
     return NetPlay::SyncIdentifierComparison::DifferentGame;
 
-  const NetPlay::SyncIdentifierComparison mismatch_result =
-      is_elf_or_dol || m_is_datel_disc ? NetPlay::SyncIdentifierComparison::DifferentGame :
-                                         NetPlay::SyncIdentifierComparison::DifferentVersion;
-
-  if (m_game_id != sync_identifier.game_id)
+  if (!is_elf_or_dol && !m_is_datel_disc && m_game_id.size() >= 4 && m_game_id.size() <= 6)
   {
-    const bool game_id_is_title_id = m_game_id.size() > 6 || sync_identifier.game_id.size() > 6;
-    return game_id_is_title_id ? NetPlay::SyncIdentifierComparison::DifferentGame : mismatch_result;
+    // This is a game ID, not a title ID
+
+    // If the region doesn't match, the games aren't the same at all.
+    // We check this first because sometimes the same game uses a different ID in different regions
+    // (in particular Japan; for instance, GOYE69, GOYP69, and GGIJ13 are used by GoldenEye: Rogue
+    // Agent), and notifying the user of different regions first is more helpful than just saying
+    // the game doesn't match
+    if (m_game_id[3] != sync_identifier.game_id[3])
+      return NetPlay::SyncIdentifierComparison::DifferentRegion;
```

Comment:
This will be confusing for users. If the user doesn't have any version of the selected netplay game, but they have *any* game that has a different fourth character from the selected game, Dolphin will say that they have the wrong version of the game instead of not having the game. I don't think that's worth handling the edge case of games that use different initial three characters in different regions.

Reply:
Changed.  Users probably will notice if one has a Japanese-region game and the other doesn't, since those tend to have actual different languages (unlike US vs Europe) so yeah, it's probably fine to not handle that edge-case.